### PR TITLE
[Website] Add release note for 0.17.0

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: _data/versions.yml [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
+  git-tag: '3cbcb7b62c2f2d02851bff837758637eb592a64b'
+  github-tag-link: 'https://github.com/apache/arrow/releases/tag/apache-arrow-0.17.0'
+  release-notes: 'https://arrow.apache.org/release/0.17.0.html'
+  mirrors: 'https://www.apache.org/dyn/closer.lua/arrow/arrow-0.17.0/'
+  tarball_name: 'apache-arrow-0.17.0.tar.gz'
+  mirrors-tar: 'https://www.apache.org/dyn/closer.lua/arrow/arrow-0.17.0/apache-arrow-0.17.0.tar.gz'
+  java-artifacts: 'http://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.arrow%22%20AND%20v%3A%220.17.0%22'
+  asc: 'https://downloads.apache.org/arrow/arrow-0.17.0/apache-arrow-0.17.0.tar.gz.asc'
+  sha256: 'https://downloads.apache.org/arrow/arrow-0.17.0/apache-arrow-0.17.0.tar.gz.sha256'
+  sha512: 'https://downloads.apache.org/dist/arrow/arrow-0.17.0/apache-arrow-0.17.0.tar.gz.sha512'
```
Comment (kou) [inline _data/versions.yml:31]: `/dist` is needless.
Our script may have a bug.

File: _data/versions.yml [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
-  sha512: 'https://downloads.apache.org/arrow/arrow-0.16.0/apache-arrow-0.16.0.tar.gz.sha512'
+  number: '0.17.0'
+  pinned_number: '0.17.*'
+  date: '20 April 2020'
+  git-tag: '3cbcb7b62c2f2d02851bff837758637eb592a64b'
+  github-tag-link: 'https://github.com/apache/arrow/releases/tag/apache-arrow-0.17.0'
+  release-notes: 'https://arrow.apache.org/release/0.17.0.html'
+  mirrors: 'https://www.apache.org/dyn/closer.lua/arrow/arrow-0.17.0/'
+  tarball_name: 'apache-arrow-0.17.0.tar.gz'
+  mirrors-tar: 'https://www.apache.org/dyn/closer.lua/arrow/arrow-0.17.0/apache-arrow-0.17.0.tar.gz'
```
Comment (kou) [inline _data/versions.yml:27]: Wow!
The CGI is implemented by Lua!

File: _data/versions.yml (hunk 1 above)
Comment (kszucs) [reply _data/versions.yml]: Fixed here and creating a PR to fix it in the release script.

File: _data/versions.yml (hunk 2 above)
Comment (kszucs) [reply _data/versions.yml:27]: Yep :)

**Pull request merged**